### PR TITLE
[IMP] base_tier_validation: notification hook

### PR DESCRIPTION
**Pull request by jarroyomorales**
This PR adds a hook so everyone can customize the notification subtype they prefer. While I understand this commit https://github.com/OCA/server-ux/commit/b9dc12fc8496f53f58ae9c663d6210a55430af24 , we were used to the notifications and now we have lost this funtionality.

@LoisRForgeFlow @CasVissers @etobella

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I think a new subtype should be created for this and be internal and default.

**Review by etobella — APPROVED** [review]
It does not change the current behavior but allows changes. LGTM

**Review by etobella — CHANGES_REQUESTED** [review]
I agree with @pedrobaeza proposal, much more cleaner, you can maintain the hook if you like

**Comment by jarroyomorales** [comment]
@etobella @pedrobaeza Ok, I also agree its a good option. I will keep the hook anyways.

**Review by etobella — APPROVED** [review]

**Comment by CasVissers-360ERP** [comment]
Wouldn't it be better to have an accepted and rejected subtype?

EDIT:
Like the change!

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by LoisRForgeFlow** [comment]
Let's hold on the merge until @CasVissers 's comment gets answered.

**Comment by jarroyomorales** [comment]
@CasVissers @LoisRForgeFlow @etobella If no one is against it I can separate the subtypes, since at the end it doesnt do any harm...

**Comment by CasVissers-360ERP** [comment]
@jarroyomorales 
I would prefer to separate the subtypes, makes it as flexible as possible without great complexity.

**Review by CasVissers-360ERP — APPROVED** [review]
👍

**Comment by jarroyomorales** [comment]
Can this get merged?

**Comment by pedrobaeza** [comment]
/ocabot merge minor

To be ported to v13

**Comment by pedrobaeza** [comment]
Ouch, there was a hidden error that has been discovered due to https://github.com/OCA/maintainer-quality-tools/pull/672. We need to fix it first before being able to merge.

**Review by etobella — CHANGES_REQUESTED** [review]
With this change, it is required to set mail as a required element on base_tier_validation. Install / upgrade might not work properly

**Comment by pedrobaeza** [comment]
Indeed, `mail` dependency is not set in manifest.

**Comment by LoisRForgeFlow** [comment]
Sorry, I didn't refresh the page and didn't see latest commits, I aborte merge operation.

**Comment by LoisRForgeFlow** [comment]
Regarding the problem, we have been avoiding the mail dependency for a long time but I guess it is not possible anymore... anyway I cannot really imaging an Odoo instance without mail installed.

**Comment by jarroyomorales** [comment]
@LoisRForgeFlow @pedrobaeza @etobella You are right about the `mail` dependency. I have just added it.

**Comment by pedrobaeza** [comment]
@LoisRForgeFlow see my comment in https://github.com/OCA/server-ux/pull/174#issuecomment-652531099

**Comment by LoisRForgeFlow** [comment]
Oh, I saw that PR merged and I assumed it was fixed.

**Comment by jarroyomorales** [comment]
> Ouch, there was a hidden error that has been discovered due to [OCA/maintainer-quality-tools#672](https://github.com/OCA/maintainer-quality-tools/pull/672). We need to fix it first before being able to merge.

This was already merged before the first attempt of merging this, so I don't know where the error might be coming from...

**Comment by pedrobaeza** [comment]
hehe, the linked PR is merged and what does is to enable the skipped tests that were before, which means that now some branches will become red because their tests were not executed, which is the case of this branch. It's not your fault, but fixing the other module test is mandatory for being able to merge this through the bot.

**Comment by etobella** [comment]
But this has no sense. The travis of the PR should fail, not only on the bot merging :thinking: 
Also, I made the installation of the same PR locally and it works as expected

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:

**Comment by LoisRForgeFlow** [comment]
thanks god my fix (do nothing) worked! :rofl:

**Pull request merged**
